### PR TITLE
Stronger invariants in unification signature.

### PR DESCRIPTION
**Pull request by ppedrot**
We use an option type instead of returning a pair with a boolean. Indeed, the boolean being true was always indicating that the returned value was unchanged.

The previous API was somewhat error-prone, and I don't understand why it was designed this way in the first place.

### CODE REVIEW

**Review by SkySkimmer — APPROVED** [review]
yes

File: pretyping/reductionops.ml [hunk 1]
```diff
@@ -1348,11 +1348,10 @@ let infer_conv_gen conv_fun ?(catch_incon=true) ?(pb=Reduction.CUMUL)
     ?(ts=full_transparent_state) env sigma x y =
   (** FIXME *)
   try
-    let b, sigma = 
-      let ans =
-	if pb == Reduction.CUMUL then
```
Comment (gares) [inline pretyping/reductionops.ml]: unrelated

**Review by gares — COMMENTED** [review]
if only `option A` was not a monad...

**Review by gares — APPROVED** [review]

**Review by mattam82 — APPROVED** [review]
I don't know why it was implemented using a boolean either, and I prefer the option version.

**Comment by mattam82** [comment]
I think this deserves a `changes.md` entry before I merge.

**Comment by ppedrot** [comment]
@mattam82 Done.

**Comment by mattam82** [comment]
I assume we had a clean CI result at some point. I don't see how this could have introduced a non-uniform bug.
